contracts/Staking.sol

```diff
@@ -109,7 +108,16 @@ contract Staking {
         }
 
         // decrease the balance this user contributed to the poolSize at the beginning of the current epoch == the epoch balance of previous epoch
-        poolSize[tokenAddress][currentEpoch].size = poolSize[tokenAddress][currentEpoch].size.sub(getEpochUserBalance(msg.sender, tokenAddress, currentEpoch));
+        uint256 epochStartBalance = getEpochUserBalance(msg.sender, tokenAddress, currentEpoch);
+        uint256 epochBalanceLeft;
+        if (amount >= epochStartBalance) {
+            epochBalanceLeft = 0;
+        } else {
+            epochBalanceLeft = epochStartBalance - amount;
```

Comment:
please use safemath

Reply:
I thought it's not really necessary because we already check that `amount is >= epochStartBalance`